lib/active_record/events/naming.rb

```diff
@@ -6,10 +6,15 @@ class Naming
       def initialize(infinitive, options = {})
         @infinitive = infinitive
         @object = options[:object].presence
+        @field_type = options[:field_type]
       end
 
       def field
-        [@object, past_participle, 'at'].compact.join('_')
+        if @field_type == :date
+          [@object, past_participle, 'on'].compact.join('_')
```

Comment:
There's a lot of similar code in both branches of the `if` statement, we could reduce it by extracting a local variable:
```ruby
suffix = @field_type == :date ? 'on' : 'at'
```